router/router.go

```diff
@@ -637,15 +637,23 @@ func (rt *HandleT) generatorLoop() {
 
 		//#JobOrder (See comment marked #JobOrder
 		rt.toClearFailJobIDMutex.Lock()
+		toDeleteIdxs := []int{}
 		for idx := range rt.toClearFailJobIDMap {
 			wrk := rt.workers[idx]
+			if len(wrk.channel) > 0 {
```

Comment:
can it happen that a user's events will be blocked forever as wrk.channel is never empty? Scenario: Only one user events are failing and everyone else's events are going through.